CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.15...3.26)
+
+project(${SKBUILD_PROJECT_NAME})
+
+set(PYBIND11_NEWPYTHON ON)
+
+find_package(
+  Python
+  COMPONENTS Interpreter Development.Module NumPy
+  REQUIRE
+)
+
+find_package(pybind11 CONFIG REQUIRED)
+
+if (TILEDB_PATH)
+    # If TILEDB_PATH variable is present prepend it to the CMAKE_PREFIX_PATH
+    # so that find_package will be able to find TileDB core library there
+    set(CMAKE_PREFIX_PATH "${TILEDB_PATH};${CMAKE_PREFIX_PATH}")
+endif()
+
+
+# In the future we should use vcpkg
+find_package(TileDB QUIET)
+if (NOT TileDB_FOUND)
+    include(cmake/DownloadTileDB.cmake)
+    if (TILEDB_VERSION)
+        message(STATUS "Downloading TileDB version \"${TILEDB_VERSION}\" ...")
+        if (TILEDB_HASH)
+            fetch_prebuilt_tiledb(
+                    VERSION ${TILEDB_VERSION}
+                    RELLIST_HASH SHA256=${TILEDB_HASH}
+            )
+        else()
+            fetch_prebuilt_tiledb(
+                    VERSION ${TILEDB_VERSION}
+            )
+        endif()
+    else()
+        message(STATUS "Downloading TileDB default version ...")
+        # Download latest release
+        fetch_prebuilt_tiledb(
+                VERSION 2.24.0
```

Comment:
Follow-up: we need to move this out at least to the top of the cmakelist if not the workflow file.

Reply:
Also I'm not sure if this is being validated correctly, because this is the has for 2.24.2

Reply:
Nevermind, it is definitely being validated 😆 (I grabbed the wrong one)

Reply:
What about a variable named TILEDB_VERSION? This variable can set on the top of the file